src/main/webapp/app/app-routing.module.ts

```diff
@@ -14,6 +14,7 @@ const LAYOUT_ROUTES: Routes = [navbarRoute, ...errorRoute];
                     path: 'admin',
                     loadChildren: () => import('./admin/admin.module').then(m => m.ArtemisAdminModule),
                 },
+                { path: 'code-editor', loadChildren: () => import('./code-editor/code-editor.module').then(m => m.ArtemisCodeEditorModule) },
```

Comment:
This is how lazy loading is configured.

Reply:
I guess to fully profit from lazy loading, we would also need to define in web pack that the code is shipped in a different chunk than the main chunk, right?

Reply:
https://ultimatecourses.com/blog/lazy-loading-angular-code-splitting-webpack describes how to do that with webpack configurations. Did you follow such an approach?

I think the admin module also uses the notation above, but I don't think it will be lazy loaded at the moment.

Reply:
Ok, I just read in the PR description that this works for the code-editor module.
Could you also make this work for the admin module? (should be easy as it is already prepared for it, but somehow does not work yet) 

The admin module will only be loaded by admin users which is less than 0,1% of our user base. Even if it's small, it will improve the initial loading time ;-)

Reply:
You are right, the admin module was not lazy loaded.
The reason was that it was still listed as an import in the app.module.

I removed it, now the chunk will be loaded when the user opens the admin page for the first time. I also added a small dummy jest test so that we can verify the admin does not have any dependency related issues, see explanation from description:

_:information_source: This also changes the behavior of the client in case of a module definition error in the code-editor module; this would only be detected when the bundle is loaded (atm we load everything at the start, so the app can't be opened if there is a module error somewhere). This issues are caught by our tests though, so we shouldn't make modules lazy that don't have tests._

Reply:
![image](https://user-images.githubusercontent.com/28230611/63676578-8985cc80-c7eb-11e9-9a74-1d538e00d0b4.png)

(The code-editor loads 2 chunks, one for the code-editor itself and one for the tree view module for the file browser)